### DISCUSSION TITLE
Verify password for an account from params.

### DISCUSSION THREAD

**Discussion by pozzos** [janko/rodauth-rails#224] [Q&A]
Hello!

This is no doubt a daft question, but I can't seem to find the correct method to verify that a given string is the correct password for an account. The context is that I have a settings with a panel of sensitive actions, and when receiving the JSON request I am including a param "password", which I would like to check before proceeding to the destructive action (in case someone has left a computer logged in unattended). I imagine there ought to be something like valid_password? that Devise has, but I haven't found anything.

Any help would be most gratefully appreciated.

Best regards,
Sebastian

**Comment by janko** [comment]
Hello, you're probably looking for the `rodauth.password_match?` method, which you can call from your controller:

```rb
unless rodauth.password_match?(params[:password])
  # ...
end
```

Note that you can also use the `confirm_password` feature coupled with `password_grace_period`, and call `rodauth.require_password_authentication` before sensitive actions. On the client side you can detect the password required response and serve a password dialog to the user, then submit the password to the `/confirm-password` action. Once the password is successfully confirmed, you can request the sensitive action again, and `rodauth.require_password_authentication` will detect the password was recently confirmed, so it will let the request through. You can set the error reason with `set_error_reason { |reason| json_response[:error_reason] = reason }` and match `password_authentication_required` on the client side.
